### ISSUE TITLE
Update <SiteNav /> to conform to a WCAG treeview specification

### ISSUE THREAD

**Issue by charrondev** [vanilla/vanilla#9091]
Previous investigation task https://github.com/vanilla/knowledge/issues/435

See specification here https://www.w3.org/TR/wai-aria-practices/examples/treeview/treeview-2/treeview-2a.html

Most of the keyboard shortcuts are already defined here: https://github.com/vanilla/vanilla/blob/ee11fde272443b635ae3892455861f7646bfc296/library/src/scripts/navigation/SiteNav.tsx#L122

## The trouble

All of the existing keyboard definitions have been defined using our tab handler. According to the treeview spec, the treeview should implement a roving tab index, meaning that only 1 item of the tree should be tabbable at a time from the rest of the page. Fixing this will require:

- Implement roving tab-index.
- Re-implement the keyboard events to not care if something is tabbable.

## Proposed implementation options:

1. Calculate which items need to be focused next by manipulating the structured data. Some tree walking would need to be done, but isn't necessarily that complicated. New tests would be required for this logic (needs to exclude hidden items, and following the proper flat order, even though the tree is nested).
2. Use some `querySelectorAll` and filter based on item visibility. I'm not 100% sure this will return items in the correct order though. Tests will need to be done.
